chaco/plot_containers.py

```diff
@@ -321,51 +140,59 @@ def _do_layout(self):
+
+    ### Persistence ###########################################################
+
+    # PICKLE FIXME: blocked with _pickles, but not sure that was correct.
+    def __getstate__(self):
+        state = super(HStackedContainer, self).__getstate__()
+        if "stack_index" in state:
+            del state["stack_index"]
+        return state
 
 
-class VPlotContainer(StackedPlotContainer):
+class VPlotContainer(VStackedContainer):
     """
     A plot container that stacks plot components vertically.
     """
 
-    draw_order = Instance(list, args=(DEFAULT_DRAWING_ORDER,))
+    #: Redefine the container layers to name the main layer as "plot" instead
+    #: of the Enable default of "mainlayer"
+    container_under_layers = Tuple("background", "image", "underlay", "plot")
 
-    #: Overrides StackedPlotContainer.
-    stack_dimension = "v"
-    #: Overrides StackedPlotContainer.
-    other_dimension = "h"
-    #: Overrides StackedPlotContainer.
-    stack_index = 1
+    #: Redefine the draw layer to "plot" instead of "mainlayer"
+    draw_layer = Str("plot")
 
-    # VPlotContainer attributes
+    #: Redefine the draw order
+    draw_order = Instance(list, args=(DEFAULT_DRAWING_ORDER,))
 
-    #: The horizontal alignment of objects that don't span the full width.
-    halign = Enum("left", "right", "center")
+    def get_preferred_size(self, components=None):
+        """Returns the size (width,height) that is preferred for this component.
 
-    #: The order in which components in the plot container are laid out.
-    stack_order = Enum("bottom_to_top", "top_to_bottom")
+        Overrides PlotComponent.
+        """
+        if self.fixed_preferred_size is not None:
+            self._cached_preferred_size = self.fixed_preferred_size
+            return self.fixed_preferred_size
 
-    #: The amount of space to put between components.
-    spacing = Float(0.0)
+        if self.resizable == "":
+            self._cached_preferred_size = self.outer_bounds[:]
+            return self.outer_bounds
 
-    def _do_layout(self):
-        """Actually performs a layout (called by do_layout())."""
-        if self.stack_order == "bottom_to_top":
-            components = self.components
-        else:
-            components = self.components[::-1]
-        if self.halign == "left":
-            align = "min"
-        elif self.halign == "center":
-            align = "center"
-        else:
-            align = "max"
+        return stacked_preferred_size(self, components=components)
+
+    ### Persistence ###########################################################
 
-        return self._do_stack_layout(components, align)
+    # PICKLE FIXME: blocked with _pickles, but not sure that was correct.
+    def __getstate__(self):
+        state = super(VStackedContainer, self).__getstate__()
+        if "stack_index" in state:
+            del state["stack_index"]
+        return state
```

Comment:
If we can push this `__getstate__` down into enable, we can get rid of these in the chaco subclasses. Note that this PR also removes a bunch of traits in the subclasses which were marked as `transient` in the subclasses but they are not `transient` in the superclasses e.g. `_cached_preferred_size` (which isn't even defined on the baseclasses in the first place)